Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,251 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Claus Clausen, Patrick Massot
+-/
+
+import Mathlib.Analysis.SpecialFunctions.Gaussian
+import Mathlib.Probability.Notation
+import Mathlib.Probability.Cdf
+
+/-! # Exponential distributions over ℝ
+
+Define the Exponential Measure over the Reals
+
+## Main definitions
+* `exponentialPdfReal`: the function `r x ↦ r * (Real.exp (-(r * ↑x))` for `0 ≤ x`
+  or `0` else, which is the probability density function of a exponential distribution with
+  rate `r` (when `hr : 0 < r`).
+* `exponentialPdf`: `ℝ≥0∞`-valued pdf,
+  `exponentialPdf r = ENNReal.ofReal (exponentialPdfReal r)`.
+* `expMeasure`: an exponential measure on `ℝ`, parametrized by its rate `r`.
+* `exponentialCdfReal`: the Cdf given by the Definition of CDF in `ProbabilityTheory.Cdf` on
+
+## Main results
+* `ExpCdf_eq`: Proof that the `exponentialCdfReal` given by the Definition equals the known
+  function given as `r x ↦ 1 - (Real.exp (-(r * ↑x))` for `0 ≤ x` or `0` else.
+-/
+
+open scoped ENNReal NNReal Real
+
+open MeasureTheory Real Set Filter Topology
+
+  /-- A Lebesgue Integral from -∞ to y can be expressed as the sum of one from -∞ to 0 and 0 to x -/
+lemma lintegral_split_bounded {y z : ℝ} (f : ℝ → ENNReal) (hzy : z ≤ y) :
+    ∫⁻ x in Iic y, f x = (∫⁻ x in Iio z, f x) + ∫⁻ x in Icc z y, f x := by
+  rw [← Iio_union_Icc_eq_Iic hzy, lintegral_union measurableSet_Icc]
+  rw [Set.disjoint_iff]
+  rintro x ⟨h1 : x < _, h2, _⟩
+  linarith
+
+lemma lintegral_split (f : ℝ → ENNReal) (c : ℝ) :
+    ∫⁻ x, f x = (∫⁻ x in {x | x ≥ c}, f x) + ∫⁻ x in {x | x < c}, f x := by
+  have union : univ = {x: ℝ | x ≥ c} ∪ {x : ℝ | x < c} := by
+    ext x; simp [le_or_lt]
+  have : IsOpen {x : ℝ | x < c} := by exact isOpen_gt' c
+  calc
+  ∫⁻ x, f x = ∫⁻ x in univ, f x ∂ volume := (set_lintegral_univ f).symm
+  _ = ∫⁻ x in {x | x ≥ c} ∪ {x | x < c} , f x ∂ volume := by rw [← union]
+  _ = _ := by
+    apply lintegral_union this.measurableSet
+    rw [Set.disjoint_iff]
+    rintro x ⟨hxge : x ≥ _, hxlt : x < _⟩
+    linarith
+
+namespace ProbabilityTheory
+
+section ExponentialPdf
+
+/-- Define the PDF of the exponential distribution depending on its rate-/
+noncomputable
+def exponentialPdfReal (r x : ℝ) : ℝ :=
+  if 0 ≤ x then r * exp (-(r * x)) else 0
+
+/-- The PDF of the exponential Distribution on the extended real Numbers-/
+noncomputable
+def exponentialPdf (r x : ℝ) : ℝ≥0∞ :=
+  ENNReal.ofReal (exponentialPdfReal r x)
+
+lemma exponentialPdf_eq (r x : ℝ) :
+    exponentialPdf r x = ENNReal.ofReal (if 0 ≤ x then r * exp (-(r * x)) else 0) := rfl
+
+lemma antiDeriv_expDeriv_pos' {r x : ℝ} (hr : 0 < r) :
+    HasDerivAt (fun a ↦ -1/r * exp (-(r * a))) (exp (-(r * x))) x := by
+  convert (((hasDerivAt_id x).const_mul (-r)).exp.const_mul (-1/r)) using 1 <;> field_simp
+
+/-- the Lebesgue-Integral of the exponential PDF over nonpositive Reals equals 0-/
+lemma lintegral_nonpos {x r : ℝ} (hx : x ≤ 0) :
```

Comment:
This name is way to generic: it's not about any Lebesgue integral. Also when a part of the name refers to a hypothesis (like `nonpos` here), we prefix it with `of`:
```suggestion
lemma lintegral_exponentialPdf_of_nonpos {x r : ℝ} (hx : x ≤ 0) :
```